services/src/main/scala/cromwell/services/metadata/impl/archiver/ArchiveMetadataConfig.scala

```diff
@@ -0,0 +1,47 @@
+package cromwell.services.metadata.impl.archiver
+
+import akka.actor.ActorSystem
+import com.typesafe.config.Config
+import common.Checked
+import common.validation.Validation._
+import cromwell.core.filesystem.CromwellFileSystems
+import cromwell.core.path.PathFactory.PathBuilders
+import cromwell.core.path.{PathBuilderFactory, PathFactory}
+import cromwell.core.{WorkflowId, WorkflowOptions}
+import net.ceedubs.ficus.Ficus._
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+import scala.language.postfixOps
+import scala.util.Try
+
+final case class ArchiveMetadataConfig(pathBuilders: PathBuilders,
+                                       bucket: String,
+                                       bucketReadLimit: Int,
+                                       interval: FiniteDuration,
+                                       debugLogging: Boolean) {
+  def makePath(workflowId: WorkflowId)= PathFactory.buildPath(ArchiveMetadataConfig.pathForWorkflow(workflowId, bucket), pathBuilders)
+}
+
+object ArchiveMetadataConfig {
+
+  // TODO: Confirm if this makes sense to the users? Should we store /bucket/parent-wf-id/sub-wf-id or /bucket/wf-id ?
```

Comment:
```suggestion
  // TODO: Confirm if this makes sense to the users? Should we store /bucket/parent-wf-id/sub-wf-id or /bucket/wf-id ?
  // When deciding keep in mind that workflows can nest to arbitrary depth, mustn't exceed path length limits: /bucket/parent-wf-id/parent-wf-id/parent-wf-id/parent-wf-id/sub-wf-id
```

Reply:
LGTM. Also made https://broadworkbench.atlassian.net/browse/BW-596 so that we're less likely to forget to make an active decision on this.